gateway/handler_unixfs_dir.go

```diff
@@ -61,9 +60,15 @@ func (i *handler) serveDirectory(ctx context.Context, w http.ResponseWriter, r *
 
 	// Check if directory has index.html, if so, serveFile
 	idxPath := ipath.Join(contentPath, "index.html")
-	idx, err := i.api.GetUnixFsNode(ctx, idxPath)
+	idxResolvedPath, err := i.api.ResolvePath(ctx, idxPath)
 	switch err.(type) {
 	case nil:
+		idx, err := i.api.GetUnixFsNode(ctx, idxResolvedPath)
+		if err != nil {
+			internalWebError(w, err)
+			return
+		}
+
```

Comment:
I updated the signature of the getter for UnixFS files:

```diff
- GetUnixFsNode(context.Context, path.Path) (files.Node, error)
+ GetUnixFsNode(context.Context, path.Resolved) (files.Node, error)
```

We already have `ResolvePath` at our disposal. Let's simplify. The change in this code is to reflect that.